### DISCUSSION TITLE
MPxSubsceneOverride and GL (Hydra)

### DISCUSSION THREAD

**Discussion by ChokmahBinah** [Autodesk/maya-usd#2176] [General]
Hello,

I have created a custom node node using MPxSubsceneOverride() implementation.
It works really well using Viewport 2.0, but nothing appears when using GL(Hydra).

Is there a way to make this kind of custom locators display using GL (Hydra) ?

I don't know how to do for now.

Using Viewport 2.0 :
![image](https://user-images.githubusercontent.com/33164632/157071093-139ca786-bc26-4e6a-9b91-e834a14fda45.png)

Using HD (Hydra) :
![image](https://user-images.githubusercontent.com/33164632/157071266-75e29506-87f2-47ea-9d81-259a3ff74aa1.png)

Thanks in advance for your help :)

**Comment by santosd** [comment]
@williamkrick, would you happen to know where @ChokmahBinah can take a look into resolving this issue on the MtoH side? Thanks!

**Comment by williamkrick** [comment]
Hi @ChokmahBinah, the GL(Hydra) viewport only draws shape types which are supported by the mtoh plugin (see [MayaToHydra.md](https://github.com/Autodesk/maya-usd/blob/3f854b60680a6a050f3ef0390c991d516b72ef36/doc/MayaToHydra.md) for currently supported shape types). I'm not sure how exactly you are drawing the custom locators in Maya, but those nodes would have to be supported by mtoh for them to draw in the GL(Hydra) viewport.

Right now the mtoh plugin and the GL(Hydra) viewport which depends on mtoh are considered experimental by the MayaUSD team. We are not actively adding features to mtoh at this time.

You should be able to add support for the necessary nodes yourself. As a starting point the code to adapt a Maya mesh into something Hydra understands is in [meshAdapter.cpp](https://github.com/Autodesk/maya-usd/blob/dev/lib/usd/hdMaya/adapters/meshAdapter.cpp).

**Reply by ChokmahBinah** [reply]
Thank you for your reply @williamkrick ,

Yes I know mtoh is experimental, so your answer helps me on my investigations.

The plugin I am writing is using Maya python api 2.0. I'll try to read about meshAdapter.cpp to see what I could use.

Then again for your answer @williamkrick :)

**Reply by williamkrick** [reply]
@ChokmahBinah meshAdapter.cpp is a file in MayaUSD. I was thinking you'd have to modify the MayaUSD plugin & re-compile it to get it working. AFAIK there is no interface to plug in a python adapter to mtoh so I don't believe a pure python implementation is possible, but I could easily be incorrect.

Good luck!

**Reply by ChokmahBinah** [reply]
Good to know and thank again.

I'll give a look at meshAdapter.cpp anyway. I think I will learn many thing from it ;)
